File: two-pointers/75-sort-colors.py
class Solution(object):
    def sortColors(self, nums):
        """
        :type nums: List[int]
        :rtype: None Do not return anything, modify nums in-place instead.
        """
        l =0
        r = len(nums)-1
        i = 0

        while i <= r:
            if nums[i]==0:
                nums[i], nums[l] = nums[l], nums[i]
                i+=1
                l+=1
            elif nums[i]==1:
                i+=1
            else:
                nums[i], nums[r] = nums[r], nums[i]
                # i+=1
                r-=1
            print(i, nums)




# Solution().sortColors([2,0,2,1,1,0])
# Solution().sortColors([2,0,1])
Solution().sortColors([1, 2,0])

File: two-pointers/75-sort-colors.go
package main

func sortColors(nums []int) {
	l, r := 0, len(nums)-1
	i := 0
	for i <= r {
		if nums[i] == 0 {
			nums[l], nums[i] = nums[i], nums[l]
			i++
			l++

		} else if nums[i] == 1 {
			i++

		} else {
			nums[r], nums[i] = nums[i], nums[r]
			r--
		}
	}
}
func main() {

	sortColors([]int{2, 0, 2, 1, 1, 0})
}
